docker-compose.yml

```diff
@@ -36,10 +38,46 @@ services:
       context: .
       dockerfile: Dockerfile
       target: testing
-    image: todo_app:test
+    image: todo-app:test
     volumes:
       - type: bind
         source: ./todo_app
         target: /app/todo_app
     profiles:
       - test
+
+  e2e_testing:
+    build:
+      context: .
+      dockerfile: Dockerfile
+      target: e2e_testing
+    image: todo-app:e2e_test
+    volumes:
+      - type: bind
+        source: ./todo_app
+        target: /app/todo_app
+    env_file:
+      - ".env"
+    profiles:
+      - e2e_test
+    network_mode: "bridge"
+  
+  linter:
```

Comment:
As mentioned above, this says it's linting but is actually going a little beyond and formatting the code too - which is great but possibly confusing (and certainly made it non-obvious to me why we were overriding the relevant command in your pipeline!)

One option we could follow is explicitly calling those out, and using the `extends` keyword to minimise the extra compose setup. E.g. we could add:
```yml
  check-style:
    extends: linter
    command: "poetry run pycodestyle --max-line-length=200 todo_app tests e2e_tests"
    profiles:
      - lint
```

(We could similarly have linter extend dev potentially!)